src/xrpld/app/tx/detail/AMMClawback.cpp

```diff
@@ -57,10 +58,19 @@ AMMClawback::preflight(PreflightContext const& ctx)
 
     std::optional<STAmount> const clawAmount = ctx.tx[~sfAmount];
     auto const asset = ctx.tx[sfAsset];
+    auto const asset2 = ctx.tx[sfAsset2];
 
     if (isXRP(asset))
         return temMALFORMED;
 
+    if (flags & tfClawTwoAssets && asset.account != asset2.account)
+    {
+        JLOG(ctx.j.trace())
+            << "AMMClawback: tfClawTwoAssets can only be enabled when two "
+               "assets in the AMM pool are both issued by the issuer";
+        return temMALFORMED;
```

Comment:
I used temINVALID_FLAG in my first commit, @gregtatcam adviced to change from temINVALID_FLAG to temMALFORMED.

Reply:
This is what the requestor of this issue suggested. I'm fine with `temINVALID_FLAG`. How about `temBAD_ISSUER`? After all this is what it's pretty much is.

Reply:
But if we change this to `temBAD_ISSUER` then I think we should change to the same error this check

    if (asset.account != issuer)

Reply:
I have to disagree here. We really should get away from using what I've been calling "redundant" error codes. `tem` _means_ malformed, so `temMALFORMED` means "malformed malformed". (And before anyone points it out, I know it's used in a ton of places in the existing code, but that doesn't mean it's correct here.) This is especially true when there are other good options that already exist.
* `temINVALID_FLAG` has already been mentioned. It makes sense because you're not allowed to use that flag when the AMM has assets from two different issuers. One way to look at it is that `tfClawTwoAssets` is a flag defined for a very specific special case - when the AMM has two different assets from the same issuer _and_ the issuer wants to get both of them, instead of the default behavior of just one. Thus, it is *invalid* to use the *flag* when the issuers are different. 
   * The two assets are properties of the AMM. They're not going to change. If the AMM was created with assets from two different issuers, then there is nothing you can change about the transaction to work with that AMM and use `tfClawTwoAssets`.
* `temBAD_AMM_TOKENS` could be considered, though the error message would need to be updated. The transaction can't be processed with the given flags because the tokens are not compatible with it.
* `temBAD_ISSUER` is another option. The flag can not be used with the issuer of `asset2`.
* `temINVALID_ACCOUNT_ID` is a stretch, but it could be considered invalid to specify that an `asset2` account ID that is not the same as `asset`.

Of these four options, though, I think `temINVALID_FLAG` is the best, and is better than `temMALFORMED` because at least it communicates _some_ information about the problem.

Reply:
OK. Since everyone agrees on `temINVALID_FLAG`, I changed the return error to `temINVALID_FLAG`

Reply:
@gregtatcam From my perspective, the flag can only be set if the issuer of the two assets are the same. So the flag would be considered as "invalid"  if the issuers are different. I believe I did something similar in [MPTokenIssuanceSet](https://github.com/XRPLF/rippled/blob/0ec17b6026298dc150099b66a2cecad0fe561d1b/src/xrpld/app/tx/detail/MPTokenIssuanceSet.cpp#L43), where I returned `temINVALID_FLAG` if both flags are set. I don't believe there is a written rule that says we can only return this error code when the flag mask is invalid.

And we may not want to introduce additional error codes like `temBAD_ISSUER`  since there is an upper limit to the total number of err codes. 

Just my 2 cents

Reply:
Sounds good. 

`temBAD_ISSUER` is an existing error.